cluster-essentials/pkg/client/install.go

```diff
@@ -20,28 +24,65 @@ var appplyResourcesFromManifests = appplyResourcesFromManifest
 const defaultBundleDir = "/tmp"
 const defaultTimeout = 15 * time.Minute
 
+var clusterEssentialPackages = []string{"kapp-controller", "secretgen-controller"}
+
 func validatePreInstall(restConfig *rest.Config) error {
-	// TODO implementation
-	return nil
+	clientset, _ := kubernetes.NewForConfig(restConfig)
+	nsList, _ := clientset.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
```

Comment:
Please take context as an arg to the function.

Reply:
Also use controller-runtime's `Client` for CRUD operations on Kubernetes objects. client-go will bring in a bunch of unnecessary dependencies.